src/dispatcher/index.ts

```diff
@@ -100,35 +105,33 @@ const createDispatcher = () => {
   })();
 
   return {
-    _setLang: (l: string) => (lang = l),
+    _setLang: (l: string) => {
+      lang = l;
+    },
     _addToQueue: (type: HeadType, payload: MetaPayload | string): void => {
       processQueue();
 
-      switch (type) {
-        case 'title':
-          titleQueue.splice(currentTitleIndex++, 0, payload as string);
-          break;
-
-        case 'titleTemplate':
-          titleTemplateQueue.splice(
-            currentTitleTemplateIndex++,
-            0,
-            payload as string
-          );
-          break;
-
-        case 'meta':
-          metaQueue.splice(currentMetaIndex++, 0, payload as MetaPayload);
-          break;
-
-        default:
-          linkQueue.push(payload);
+      if (type === TITLE) {
+        titleQueue.splice(currentTitleIndex++, 0, payload as string);
+      } else if (type === TEMPLATE) {
+        titleTemplateQueue.splice(
+          currentTitleTemplateIndex++,
+          0,
+          payload as string
+        );
+      } else if (type === META) {
+        metaQueue.splice(currentMetaIndex++, 0, payload as MetaPayload);
+      } else {
+        linkQueue.push(payload);
       }
     },
     _removeFromQueue: (type: HeadType, payload: MetaPayload | string) => {
-      if (type === 'title') {
-        titleQueue.splice(titleQueue.indexOf(payload as string), 1);
-        if (!isServerSide)
+      if (type === TITLE || type === TEMPLATE) {
```

Comment:
CC @gustavpursche added some cases here to fallback when we remove title or template on index 0